### ISSUE TITLE
Cannot publish files with space in filename

### ISSUE THREAD

**Issue by robsyme** [nextflow-io/nextflow#4778]
## Bug report 

When publishing a file or directory that contains a space in the path, nextflow fails to publish the file because it first converts it into a URI [here](https://github.com/nextflow-io/nextflow/blob/4debd56e4ee425d7d1766b9015ad14b2ed5b0a00/plugins/nf-amazon/src/main/nextflow/cloud/aws/nio/S3Path.java#L418).

### Steps to reproduce the problem

```bash
nextflow run nf-core/rnaseq --outdir 's3://my-output-bucket/projects/robsyme/rnaseq-results'
```
### Program output 

```
Feb.-26 18:23:14.531 [PublishDir-3] WARN  nextflow.processor.PublishDir - Failed to publish file: s3://my-output-bucket/work/84/cd1cf8f30e72cd61d1cbff9db3a78d/RAP1_UNINDUCED_REP2; to: s3://my-output-bucket/projects/robsyme/rnaseq-results/star_salmon/qualimap/RAP1_UNINDUCED_REP2 [copy] -- See log file for details
java.lang.IllegalArgumentException: Illegal character in path at index 126: s3:///my-output-bucket/projects/robsyme/rnaseq-results/star_salmon/qualimap/RAP1_UNINDUCED_REP2/images_qualimapReport/Coverage Profile Along Genes (High).png
	at java.base/java.net.URI.create(URI.java:906)
	at nextflow.cloud.aws.nio.S3Path.toUri(S3Path.java:418)
	at nextflow.file.CopyMoveHelper$1.visitFile(CopyMoveHelper.java:133)
	at nextflow.file.CopyMoveHelper$1.visitFile(CopyMoveHelper.java:108)
	at java.base/java.nio.file.Files.walkFileTree(Files.java:2811)
	at nextflow.file.CopyMoveHelper.copyDirectory(CopyMoveHelper.java:140)
	at nextflow.file.FileHelper.copyPath(FileHelper.groovy:948)
	at nextflow.processor.PublishDir.processFileImpl(PublishDir.groovy:496)
	at nextflow.processor.PublishDir.processFile(PublishDir.groovy:385)
	at nextflow.processor.PublishDir.safeProcessFile(PublishDir.groovy:362)
	at jdk.internal.reflect.GeneratedMethodAccessor208.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at org.codehaus.groovy.reflection.CachedMethod.invoke(CachedMethod.java:343)
	at groovy.lang.MetaMethod.doMethodInvoke(MetaMethod.java:328)
	at groovy.lang.MetaClassImpl.doInvokeMethod(MetaClassImpl.java:1333)
	at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1088)
	at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1007)
	at org.codehaus.groovy.runtime.InvokerHelper.invokePogoMethod(InvokerHelper.java:645)
	at org.codehaus.groovy.runtime.InvokerHelper.invokeMethod(InvokerHelper.java:628)
	at org.codehaus.groovy.runtime.InvokerHelper.invokeMethodSafe(InvokerHelper.java:82)
	at nextflow.processor.PublishDir$_apply1_closure2.doCall(PublishDir.groovy:337)
	at nextflow.processor.PublishDir$_apply1_closure2.call(PublishDir.groovy)
	at groovy.lang.Closure.run(Closure.java:505)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:833)
Caused by: java.net.URISyntaxException: Illegal character in path at index 126: s3:///my-output-bucket/projects/robsyme/rnaseq-results/star_salmon/qualimap/RAP1_UNINDUCED_REP2/images_qualimapReport/Coverage Profile Along Genes (High).png
	at java.base/java.net.URI$Parser.fail(URI.java:2974)
	at java.base/java.net.URI$Parser.checkChars(URI.java:3145)
	at java.base/java.net.URI$Parser.parseHierarchical(URI.java:3227)
	at java.base/java.net.URI$Parser.parse(URI.java:3175)
	at java.base/java.net.URI.<init>(URI.java:623)
	at java.base/java.net.URI.create(URI.java:904)
	... 28 common frames omitted
```

### Environment 

* Nextflow version: master (compiled from source)
* Java version: openjdk version "17.0.6" 2023-01-17 LTS
* Operating system: macOS

**Comment by robsyme** [comment]
Perhaps we should be URL-encoding the `parts` before joining them into a URI?

**Comment by stevekm** [comment]
file names with spaces?? that is absurd, maybe it should not be supported .... lol

**Comment by pditommaso** [comment]
Worth mentioning this only happens when the trace logging is enabled, because it tries to render the path to a URI object and then to a string. 

https://github.com/nextflow-io/nextflow/blob/bb5c4f9d1870e17e23441f2d721fe86027eb6834/modules/nf-commons/src/main/nextflow/file/CopyMoveHelper.java#L134-L134

**Comment by pditommaso** [comment]
Solved by b74c0227
